web-bluetooth/read-characteristic-value-changed.js

```diff
@@ -1,13 +1,38 @@
+var bluetoothDevice;
 var batteryLevelCharacteristic;
 
 function onReadBatteryLevelButtonClick() {
-  log('Requesting Bluetooth Device...');
-  navigator.bluetooth.requestDevice(
-    {filters: anyDevice(), optionalServices: ['battery_service']})
-  .then(device => {
-    log('Connecting to GATT Server...');
-    return device.gatt.connect();
+  requestDevice()
+  .then(connectDeviceAndCacheCharacteristics)
+  .then(_ => {
```

Comment:
A nit: `connectDeviceAndCacheCharacteristics` returns `undefined`, so `()` is more accurate as a function parameter than `_`.